kafka/README.md

```diff
@@ -69,7 +69,7 @@ This object contains information about the message representation in Kafka.
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] | The message key.
+<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] / [AVRO file](https://github.com/asyncapi/avro-schema-parser) | The message key.
```

Comment:
🤔 I don't think `Avro file` is the allowed value here. This is saying that you can do this:

```yaml
key: 'path/to/user-create.avsc#/UserCreate'
```

Instead, what you can do here is provide a Reference Object that will point to an Avro file:

```yaml
key:
  $ref: 'path/to/user-create.avsc#/UserCreate'
```

I think you're going through a rabbit hole here: coming up with a way to reference other types of schemas. @magicmatatjahu already created an issue about that: https://github.com/asyncapi/spec/issues/622.

Reply:
Hi @fmvilas ; i don't really understand what you are expected in the bindings' doc ; i update the `kafka/json_schemas/message.json ` with the expected value as `$ref` like 
```
key:
  $ref: 'path/to/user-create.avsc#/UserCreate'
```

For the new supported types, you suggest to point to the `$ref` way (https://asyncapi.io/docs/specifications/v2.1.0#referenceObject) ? With not specially an avro file format but any value in the `$ref` value ?

Concerning the @magicmatatjahu ; and due to the proposal DRAFT status, is it appropriate to document the new future way even if the implementation is not done ?

Reply:
As you wrote, my proposition is in a DRAFT/Strawman stage, so it might not be released until January or April next year (we release specs every 3-4 months). I would not change here the information that the `key` field is Schema Object or Avro file. I would just leave the information that it is Schema Object and add in the `**Note**` under the table that Schema Object for `key` can be represented with Avro format, something like that:

```md
> **NOTE**: `key` can be represented with AVRO format.
```

Reply:
It seems right for me

Reply:
Still my first comment applies. As it's defined right now you're not saying you can use `$ref`. You're saying you can do `key: path/to/avsc` and that's not what you want. You want:

```yaml
key:
  $ref: 'path/to/avsc'
```

However, given the way `$ref` works, it's going to be dereferenced into an Avro Schema object so I'd instead define it as:

```suggestion
<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] \| [AVRO Schema Object](https://avro.apache.org/docs/current/spec.html) | The message key.
```

Then, when resolving and dereferencing the `$ref`, it will become a valid definition and it also would allow you to inline the definition right there using Avro.

Reply:
is the new proposal ok for you ?